tests/ui/generic-associated-types/multiple-type-params-with-unmet-bounds.stderr

```diff
@@ -0,0 +1,27 @@
+error[E0277]: the trait bound `String: Copy` is not satisfied
+  --> $DIR/multiple-type-params-with-unmet-bounds.rs:8:29
+   |
+LL |     let _: <() as Trait>::P<String, String>;
+   |                             ^^^^^^ the trait `Copy` is not implemented for `String`
+   |
+note: required by a bound in `Trait::P`
+  --> $DIR/multiple-type-params-with-unmet-bounds.rs:2:15
+   |
+LL |     type P<T: Copy, U: Copy>;
+   |               ^^^^ required by this bound in `Trait::P`
+
+error[E0277]: the trait bound `String: Copy` is not satisfied
+  --> $DIR/multiple-type-params-with-unmet-bounds.rs:8:12
+   |
+LL |     let _: <() as Trait>::P<String, String>;
+   |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `Copy` is not implemented for `String`
+   |
+note: required by a bound in `Trait::P`
+  --> $DIR/multiple-type-params-with-unmet-bounds.rs:2:15
+   |
+LL |     type P<T: Copy, U: Copy>;
+   |               ^^^^ required by this bound in `Trait::P`
```

Comment:
I'm certain that you're aware that the output is not correct yet but I'm leaving this comment here anyways to ensure it doesn't get forgotten by accident ^^' Thanks for working on this btw!

```suggestion
LL |     let _: <() as Trait>::P<String, String>;
   |                                     ^^^^^^ the trait `Copy` is not implemented for `String`
   |
note: required by a bound in `Trait::P`
  --> $DIR/multiple-type-params-with-unmet-bounds.rs:2:15
   |
LL |     type P<T: Copy, U: Copy>;
   |                        ^^^^ required by this bound in `Trait::P`
```

Reply:
This is possibly a strange fallout of the way we do obligation deduplication in `adjust_fulfillment_errors_for_expr_obligation`. I'm inclined to say "don't worry about it", since it only triggers when you have duplicated bounds like this on the same item...

Reply:
The second error is now gone from the removal of the select_where_possible call. I believe that we have some deduplication logic gone rogue that is silencing the trait bound error for the second parameter (likely because it looks like the same predicate?). I would like to fix that, but might have to be left as a future exercise.

Reply:
I'm gonna open a separate issue for that basically uplifting https://github.com/rust-lang/rust/issues/105306#issuecomment-1693645473